skcosmo/pcovr/pcovr.py

```diff
@@ -0,0 +1,350 @@
+                    in estimating :math:`{\\mathbf{\\hat{Y}}}`
+
+    References
+        1.  S. de Jong, H. A. L. Kiers, 'Principal Covariates
+            Regression: Part I. Theory', Chemometrics and Intelligent
+            Laboratory Systems 14(1): 155-164, 1992
+        2.  M. Vervolet, H. A. L. Kiers, W. Noortgate, E. Ceulemans,
+            'PCovR: An R Package for Principal Covariates Regression',
+            Journal of Statistical Software 65(1):1-14, 2015
+    """
+
+    def __init__(
+        self,
+        mixing=0.0,
+        n_components=None,
+        regularization=1e-6,
+        tol=1e-12,
+        full_eig=False,
+        space=None,
+        lr_args={"cv": 2, "fit_intercept": False},
+    ):
+
+        self.mixing = mixing
+        self.regularization = regularization
+        self.tol = tol
+        self.space = space
+        self.lr_args = lr_args
+        self.full_eig = full_eig
+        self.n_components = n_components
+        self.whiten = False
+
+    def fit(self, X, Y, Yhat=None, W=None):
+        """
+
+        Fit the model with X and Y. Depending on the dimensions of X,
+        calls either `_fit_feature_space` or `_fit_structure_space`
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples and
+            n_features is the number of features.
+        Y : array-like, shape (n_samples, n_properties)
+            Training data, where n_samples is the number of samples and
+            n_properties is the number of properties
+        Yhat : array-like, shape (n_samples, n_properties), optional
+            Regressed training data, where n_samples is the number of samples and
+            n_properties is the number of properties. If not supplied, computed
+            by ridge regression.
+        W : array-like, shape (n_features, n_properties), optional
+            Weights of regressed training data. If not supplied, computed
+            by ridge regression.
+
+        """
+
+        X, Y = check_X_y(X, Y, y_numeric=True, multi_output=True)
+
+        if self.n_components is None:
+            self.n_components = min(X.shape)
+
+        # Sparse eigensolvers will not work when seeking N-1 eigenvalues
+        if min(X.shape) <= self.n_components:
+            self.full_eig = True
+
+        # self._check_space(X)
+
+        if Yhat is None or W is None:
+            Yhat, W = self._compute_Yhat(X, Y, Yhat=Yhat, W=W)
+
+        self._fit(X, Yhat, W)
+
+        self.mean_ = np.mean(X, axis=0)
+        self.pxy_ = self.pxt_ @ self.pty_
+        if len(Y.shape) == 1:
+            self.pxy_ = self.pxy_.reshape(
+                X.shape[1],
+            )
+            self.pty_ = self.pty_.reshape(
+                self.n_components,
+            )
+
+        self.components_ = self.pxt_.T  # for sklearn compatibility
+
+    def _fit(self, X, Yhat, W):
```

Comment:
Does such a small partition really make sense? Plus, it seems, confusion can begin if there are fit and _fit functions

Reply:
what do you mean by "small partition"?

Reply:
Splitting fit function into fit and _fit

Reply:
Good point! I combined the two in the most recent push